tilequeue/postgresql.py

```diff
@@ -1,297 +1,66 @@
-            assert host in self.host_conns_not_in_use, \
-                'Connections already in use for host: %s' % host
-            conns = self.conns_by_host[host]
-
-            if len(conns) < self.n_conn_per_host:
-                # we are short some connections
-                # create the connections that we expect to have available
-                conn_info_with_host = dict(self.conn_info, host=host)
-                for i in range(self.n_conn_per_host - len(conns)):
-                    conn = self._make_conn(conn_info_with_host)
-                    conns.append(conn)
-
-            self.host_conns_in_use.add(host)
-            self.host_conns_not_in_use.remove(host)
-            return conns
-
-    def put_conns_for_host(self, host):
-        with self.lock:
-            assert host in self.conns_by_host, 'Unknown host: %s' % host
-            assert host in self.host_conns_in_use, \
-                'Connections not in use for host: %s' % host
-
-            # check if any connections have been closed
-            # those will need to be recreated before being returned to
-            # the rotation
-            conns_to_return = []
-            conns = self.conns_by_host.pop(host)
+        with self._lock:
             try:
-                conn_info_with_host = dict(self.conn_info, host=host)
-                for conn in conns:
-                    if conn.closed:
-                        try:
-                            conn = self._make_conn(conn_info_with_host)
-                        except:
-                            print 'Error creating new connection to host: %s' \
-                                % host
-                            # When connections are fecthed for this
-                            # host again, new ones will attempt to be
-                            # created at that point
-                            continue
-                    conns_to_return.append(conn)
-            finally:
-                # always add whatever connections we have available back
-                # and restore the host connection accounting
-                self.conns_by_host[host] = conns_to_return
-                self.host_conns_in_use.remove(host)
-                self.host_conns_not_in_use.add(host)
-
-    def closeall(self):
-        with self.lock:
-            for host in self.hosts:
-                conns = self.conns_by_host[host]
-                for conn in conns:
-                    try:
-                        conn.close()
-                    except:
-                        pass
-            self.conns_by_host.clear()
-            self.host_conns_in_use.clear()
-            self.host_conns_not_in_use.clear()
-            self.hosts = []
+                pool_to_use = next(self._pool_cycle)
+                for _ in range(n_conns):
+                    conn = pool_to_use.getconn()
 
+                    register_json(conn, loads=json.loads, globally=True)
```

Comment:
A subtle, notable difference here is that I'm passing in `ujson`'s `loads` for `psycopg2` to use when parsing the data from PostgreSQL.